plugins/nordic-hid/fu-nordic-hid-cfg-channel.c

```diff
@@ -513,16 +513,11 @@ fu_nordic_hid_cfg_channel_get_bl_name(FuNordicHidCfgChannel *self, GError **erro
 			return FALSE;
 
 		/* check if not set via quirk */
-		if (self->bl_name != NULL &&
-		    strncmp(self->bl_name, (const char *)res->data, res->data_len) != 0) {
-			g_set_error(
-			    error,
-			    G_IO_ERROR,
-			    G_IO_ERROR_INVALID_DATA,
-			    "bootloader in quirk file is '%s' while the board is supporting '%s'",
-			    self->bl_name,
-			    g_strndup((const gchar *)res->data, res->data_len));
-			return FALSE;
+		if (self->bl_name != NULL) {
+			g_debug("Bootloader readout '%s' overrides bootloader from quirk '%s'",
+				g_strndup((const gchar *)res->data, res->data_len),
```

Comment:
yes, this is fine -- but g_strndup leaks the string. you'll need to do something like `g_autofree gchar *tmp = g_strndup...` to use it in the g_debug.

Reply:
Fixed. Thanks for pointing this out.
btw. Is the leak present on in the current implementation from `main` branch? We pass the pointer as format argument in the `g_set_error`, is it automatically freed in that case?

Reply:
Nah, the leak is there on the `main` branch too -- we missed that on initial review. :)

Reply:
Thanks a lot for the review :)

Reply:
I've invited you as a contributor -- you can merge this now if you want.